src/loader/bgw_interface.c

```diff
@@ -69,24 +70,30 @@ ts_bgw_db_workers_start(PG_FUNCTION_ARGS)
 	PG_RETURN_BOOL(ts_bgw_message_send_and_wait(START, MyDatabaseId));
 }
 
+/*
+ * allow superusers OR database owners to stop and restart background workers
+ * database owners must be allowed to do this in order to do pg_dump/restore
+ */
 Datum
 ts_bgw_db_workers_stop(PG_FUNCTION_ARGS)
 {
-	if (!superuser())
+	bool db_owner = pg_database_ownercheck(MyDatabaseId, GetUserId());
+	if (!superuser() && !db_owner)
```

Comment:
```suggestion
	if (!pg_database_ownercheck(MyDatabaseId, GetUserId()))
```
Isn't more simpler in that way? The `pg_database_ownercheck` also check for `superuser`. I mean will return true (bypass) the check for super users.